test/serializers/read_attribute_for_serialization_test.rb

```diff
@@ -7,20 +7,28 @@ class ReadAttributeForSerializationTest < ActiveSupport::TestCase
       class Parent < ActiveModelSerializers::Model
         attr_accessor :id
       end
+
       class Child < Parent
         attr_accessor :name
       end
+
       class ParentSerializer < ActiveModel::Serializer
         attributes :$id
 
         define_method(:$id) do
           object.id
         end
       end
+
       class ChildSerializer < ParentSerializer
         attributes :name
       end
 
+      def test_for_serializer_initialized_with_nil_object
+        child = ChildSerializer.new(nil)
+        assert_equal nil, child.read_attribute_for_serialization(:name)
```

Comment:
This is also super easy to check for in userland code.  I'd consider adding a nil check in initialize that raises an exception earlier, but honestly, there's no reason that should ever be happening.  When serializing a nil association we skip it, rather than blow up. And if you pass nil to render json: it will skip using a serializer as well.
 That is, the exception orginates in the example scenario when a nil is knowingly being passed in.  Raising an exception in initialize instead of attritbutes doesn't change that.

> For example:

> employee = Employee.first.version_at(date) # version may or may not exist on that date.
EmployeeSerializer.new(employee).attributes

If you think this still merits a code change in AMS, would you mind providing a different example?

B mobile phone

> On May 17, 2016, at 9:32 PM, Lucas Hosseini <notifications@github.com> wrote:
> 
> In test/serializers/read_attribute_for_serialization_test.rb:
> 
> >        class ChildSerializer < ParentSerializer
> >          attributes :name
> >        end
> >  
> > +      def test_for_serializer_initialized_with_nil_object
> > +        child = ChildSerializer.new(nil)
> > +        assert_equal nil, child.read_attribute_for_serialization(:name)
> The problem with "silently failing" is that it makes misbehaving code go unnoticed and debugging very hard. When stuff fails, you know there's an issue with your code, and the sooner it fails, the easier it is to fix it.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>